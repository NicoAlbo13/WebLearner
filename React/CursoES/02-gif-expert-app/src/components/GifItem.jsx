import PropTypes from 'prop-types';

export default function GifItem({title, url, id}) {
  return (
    <div className="card">
        <img src={url} alt={title} />
        {/* Take the word 'GIF' out of the title */}
        <p>{title.split(' ').map((txt)=>{
            if(txt != 'GIF') return txt;
        }).join(' ')}</p>
    </div> 
  )
}

GifItem.prototype = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
