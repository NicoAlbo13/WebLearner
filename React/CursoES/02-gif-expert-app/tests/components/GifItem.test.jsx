import { render, screen } from '@testing-library/react';
import GifItem from '../../src/components/GifItem';

describe('test on GifItem', () => {

    const title = 'This is a title'
    const url = 'http://testurl.com/image.jpg'

    test('should GifItem module should match Snapshot', () => { 
        const { container } = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
    })

    test('should use the given url and alt on image', () => { 
        render(<GifItem title={title} url={url}/>)
        // expect(screen.getByAltText(title)).toBeTruthy();
        // expect(screen.getByRole('img').src).toBe(url);

        const {src, alt} = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);

     })

     test('should contain title on the component', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
      })

})