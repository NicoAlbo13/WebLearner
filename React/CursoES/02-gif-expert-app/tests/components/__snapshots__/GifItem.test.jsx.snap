// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on GifItem should GifItem module should match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="This is a title"
      src="http://testurl.com/image.jpg"
    />
    <p>
      This is a title
    </p>
  </div>
</div>
`;
